rlpaddr/constraints.lisp

```diff
@@ -132,10 +132,9 @@
          (begin (vanishes! (shift INDEX -7))
                 (eq! ACC NONCE)
                 (eq! BIT_ACC BYTE1)
-                (if-zero (+ (~ (eq! ACC_BYTESIZE 1))
-                            (shift BIT1 -7))
-                         (eq! 1 TINY_NON_ZERO_NONCE)
-                         (vanishes! TINY_NON_ZERO_NONCE))
+                (if (and! (eq! ACC_BYTESIZE 1) (eq! (shift BIT1 -7) 0))
+                          (eq! 1 TINY_NON_ZERO_NONCE)
+                          (vanishes! TINY_NON_ZERO_NONCE))
```

Comment:
Is the normalization implicit now ?

Reply:
Well, yes and no.  The only reason you're using normalisation here as I recall is to implement a conjunction safely.  So, I've just made this explicit (which matches the spec). How exactly `go-corset` translates this doesn't really matter too much, but we can assume it does so safely.  For example, it could avoid the normalisation by converting into CNF, or it could a normalisation, etc.  

This is relevant to our discussion on field agnosticity, since my proposal also requires safe handling of nested conjunctions.  Currently it converts them all into CNF, but I'm not sure this is actually the best idea.  In particular, without aggressive simplification, it can lead to an explosion in constraints (many of which are trivially true).  I've seen this with Amelie's tic-tac-toe example.